tests/lib/rules/no-unused-services.js

```diff
@@ -0,0 +1,190 @@
+    `this.get('${propertyName}.prop');`,
+    `get(this, '${propertyName}');`,
+    `get(this, '${propertyName}.prop');`,
+    `this.getProperties('a', '${propertyName}');`,
+    `this.getProperties('a', '${propertyName}.prop');`,
+    `this.getProperties(['a', '${propertyName}']);`,
+    `this.getProperties(['a', '${propertyName}.prop']);`,
+    `getProperties(this, 'a', '${propertyName}');`,
+    `getProperties(this, 'a', '${propertyName}.prop');`,
+    `getProperties(this, ['a', '${propertyName}']);`,
+    `getProperties(this, ['a', '${propertyName}.prop']);`,
+    `const { a, b, ${propertyName} } = this;`,
+    `let { c, ${propertyName} : prop, d } = this;`,
+  ];
+}
+
+/**
+ * Generate an array of valid test cases
+ * @returns {Array}
+ */
+function generateValid() {
+  const usecases = generateUseCasesFor(SERVICE_NAME);
+  const valid = [];
+  for (const use of usecases) {
+    valid.push(
+      `class MyClass { @service('foo') ${SERVICE_NAME}; fooFunc() {${use}} }`,
+      `class MyClass { @service() ${SERVICE_NAME}; fooFunc() {${use}} }`,
+      `Component.extend({ ${SERVICE_NAME}: service('foo'), fooFunc() {${use}} });`,
+      `Component.extend({ ${SERVICE_NAME}: service(), fooFunc() {${use}} });`
+    );
+  }
+  return valid;
+}
+
+// Testing for unrelated props + some edge cases
+const unrelatedPropUses = generateUseCasesFor('unrelatedProp').join('').concat('let foo;');
+
+ruleTester.run('no-unused-services', rule, {
+  valid: generateValid(),
+  invalid: [
+    {
+      code: `class MyClass { @service('foo') ${SERVICE_NAME}; fooFunc() {${unrelatedPropUses}} }`,
+      output: null,
+      errors: [
+        {
+          message,
+          suggestions: [{ output: `class MyClass {  fooFunc() {${unrelatedPropUses}} }` }],
+          type: 'ClassProperty',
+        },
+      ],
+    },
+    {
+      code: `class MyClass { @service() ${SERVICE_NAME}; fooFunc() {${unrelatedPropUses}} }`,
+      output: null,
+      errors: [
+        {
+          message,
+          suggestions: [{ output: `class MyClass {  fooFunc() {${unrelatedPropUses}} }` }],
+          type: 'ClassProperty',
+        },
+      ],
+    },
+    {
+      code: `Component.extend({ ${SERVICE_NAME}: service('foo'), fooFunc() {${unrelatedPropUses}} });`,
+      output: null,
+      errors: [
+        {
+          message,
+          suggestions: [{ output: `Component.extend({  fooFunc() {${unrelatedPropUses}} });` }],
+          type: 'Property',
+        },
+      ],
+    },
+    {
+      code: `Component.extend({ ${SERVICE_NAME}: service('foo') });`,
+      output: null,
+      errors: [{ message, suggestions: [{ output: 'Component.extend({  });' }], type: 'Property' }],
+    },
+    {
+      code: `Component.extend({ ${SERVICE_NAME}: service(), fooFunc() {${unrelatedPropUses}} });`,
+      output: null,
+      errors: [
+        {
+          message,
+          suggestions: [{ output: `Component.extend({  fooFunc() {${unrelatedPropUses}} });` }],
```

Comment:
We also need to assert the message provided by this suggestion. The modern way to do that would be to add `'messageId': 'removeServiceInjection'` here and then in the rule file, define `removeServiceInjection` under `meta.messages`.

We could also switch to using messageIds for the error message itself too (this would replace the `getErrorMessage` function).

https://eslint.org/docs/developer-guide/working-with-rules#messageids